pydantic/_internal/_typing_extra.py

```diff
@@ -283,8 +283,10 @@ def _make_forward_ref(
         global _make_forward_ref
         try:
             res = typing.ForwardRef(arg, is_argument, is_class=is_class)  # type: ignore
-            _make_forward_ref = typing.ForwardRef  # type: ignore
-            return res
+            # we have to ignore the following two lines from coverage because
+            # `is_class` does not exist in 3.8 and it always throws `TypeError`
+            _make_forward_ref = typing.ForwardRef  # type: ignore # pragma: no cover
+            return res  # pragma: no cover
```

Comment:
Then why do we have this logic? This is only for <=3.8 anyway 🤔

Reply:
Good catch 🎯 
After talking with @adriangb, I've changed it.

Reply:
For the record: we tracked down the origin of this logic to https://github.com/pydantic/pydantic/commit/594effa279668bd955e98f1cd5c036b37d3bbd40, so it's been there from the start. That was huge commit so it's highly likely this was never needed and we've just carried it forward.